### PR TITLE
fix 4988 Unexpected behaviors for Ellipse (Curve) sockets.

### PR DESCRIPTION
**Pull request by satabol**
Sverchok 1.3.0-alpha-master, Blender 3.6.1, Windows 11
see #4988

Issue:
issue_4988_Ellipse(Curve)_sockets_002 anim

Fixed:
issue_4988_Ellipse(Curve)_sockets anim

### CODE REVIEW

**Comment by satabol** [comment]
Hi @portnov. Can you check changes in a node "Ellipse (Curve)"? I found some strange behaviors for sockets for this node and fix it (I think so))) )

File: nodes/curve/ellipse_curve_mk2.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        else:  # AC
+            list_links_remove = ["Minor Radius", "Eccentricity",]
+
+        # remove existing link if sockets are switched:
+        for socket_name in list_links_remove:
+            all_sockets_name.remove(socket_name)
+            for l in self.inputs[socket_name].links:
+                self.id_data.links.remove(l)
+            self.inputs[socket_name].hide = True
+        self.inputs[ all_sockets_name[0] ].hide = False
```
Comment (Durman) [inline nodes/curve/ellipse_curve_mk2.py:47]: Use existing Blender API, like this:
```py
self.inputs['Eccentricity'].enabled = self.mode == "AE"
```

File: nodes/curve/ellipse_curve_mk2.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        
+        self.update_sockets(context)
+
+    def process(self):
+        outputs = self.outputs
+        # return if no outputs are connected
+        if not any(s.is_linked for s in outputs):
+            return
+
+        major_radius_s = self.inputs['Major Radius'].sv_get()
```
Comment (Durman) [inline nodes/curve/ellipse_curve_mk2.py:105]: `sv_get(deepcoopy=False)` for performance

File: nodes/curve/ellipse_curve_mk2.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
+
+from math import sqrt
```
[the remark this answers is not in the document]
Comment (satabol) [reply nodes/curve/ellipse_curve_mk2.py:2]: I know this but there was old node with the same name. What I have to do for this case?

[the remark this answers is not in the document]
Comment (Durman) [reply nodes/curve/ellipse_curve_mk2.py:2]: The old node goes into the old folder. I don't see any problem here.

[the remark this answers is not in the document]
Comment (satabol) [reply nodes/curve/ellipse_curve_mk2.py:2]: Previous node name was "Ellipse" not "Ellipse (Curve)" and folder "old_nodes" has node "ellipse.py" from Generator->Ellipse.

File: nodes/curve/ellipse_curve_mk2.py (hunk 2 above)
Comment (satabol) [reply nodes/curve/ellipse_curve_mk2.py]: fixed

File: nodes/curve/ellipse_curve_mk2.py (hunk 1 above)
Comment (satabol) [reply nodes/curve/ellipse_curve_mk2.py]: fixed

**Comment by satabol** [comment]
@Durman Is it right names?

**Review by Durman — APPROVED** [review]

**Pull request merged**